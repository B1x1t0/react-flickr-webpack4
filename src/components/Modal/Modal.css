.Modal {
    position: fixed;
    top: 0;
    left: 0;
    width: 100%;
    height: 100vh;
    background: rgba(0,0,0,0.8);
    box-sizing: border-box;
    transition: all 0.3s ease-out;
}

.Close {
    height: 50px;
    color: #FFFFFF;
    font-size: 28px;
    text-align: right;
    width: 100%;
}