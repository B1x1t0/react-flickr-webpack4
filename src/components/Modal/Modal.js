import React, { Component } from 'react';
import classes from './Modal.css'
import ImageInfo from './ImageInfo/ImageInfo';


class Modal extends Component {
    render () {
        let image = '';

        if(this.props.show){
            image =  (<ImageInfo data={this.props.imageData}></ImageInfo>)
        }

        return (
            <div className={classes.Modal}
                style={{
                    transform: this.props.show ? 'translateY(0)' : 'translateY(-100vh)',
                    opacity: this.props.show ? '1' : '0'
                }}
                onClick={this.props.hide}
            >
            <div className={classes.Close} onClick={this.props.hide}>Close</div>
            {image}
        </div>
    )
    };
};

export default Modal;