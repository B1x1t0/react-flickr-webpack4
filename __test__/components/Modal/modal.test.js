import React from 'react';
import Modal from '../../../src/components/Modal/Modal';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({adapter: new Adapter()});
const mockFn = jest.fn();

describe('Modal', () => {
 it('should be defined', () => {
   expect(Modal).toBeDefined();
 });
it('should render correctly', () => {
   const tree = shallow(
     <Modal />
   );
   expect(tree).toMatchSnapshot();
 });

 it('should call function when modal is clicked', () => {
    const tree = shallow(
      <Modal hide={mockFn} />
    );
    tree.simulate('click');
    expect(mockFn).toHaveBeenCalled();
  });
});