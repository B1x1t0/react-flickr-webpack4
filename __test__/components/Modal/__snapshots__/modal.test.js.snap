// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Modal />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="Close"
          onClick={undefined}
        >
          Close
        </div>,
        "",
      ],
      "className": "Modal",
      "onClick": undefined,
      "style": Object {
        "opacity": "0",
        "transform": "translateY(-100vh)",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Close",
          "className": "Close",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "Close",
        "type": "div",
      },
      "",
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="Close"
            onClick={undefined}
          >
            Close
          </div>,
          "",
        ],
        "className": "Modal",
        "onClick": undefined,
        "style": Object {
          "opacity": "0",
          "transform": "translateY(-100vh)",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Close",
            "className": "Close",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "Close",
          "type": "div",
        },
        "",
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
